src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -240,6 +240,45 @@ public String[] findEnumValues(Class<?> enumType, Enum<?>[] enumValues, String[]
         return names;
     }
 
+    @Override
+    public String[] findEnumValues(MapperConfig<?> config, Enum<?>[] enumValues, String[] names,
+                                   AnnotatedClass annotatedClass) {
+        // First collect all JsonProperty.value()
+        HashMap<String, String> enumToPropertyMap = new HashMap<>();
+        for (AnnotatedField field : annotatedClass.fields()) {
+            JsonProperty property = field.getAnnotation(JsonProperty.class);
+            if (property != null) {
+                enumToPropertyMap.put(field.getName(), property.value());
+            }
+        }
+        final Class<?> enumType = annotatedClass.getRawType();
+        HashMap<String,String> expl = null;
+        for (Field f : enumType.getDeclaredFields()) {
```

Comment:
Why iterate over these again? Didn't `AnnotatedClass.fields()` have everything we need?

Reply:
Oh, you are right about that `AnnotatedClass.fields()` does and "should" have everything we need. This one was from previous implementation and no longer needed.

Removed, thankssss